testdata/p4tc_samples_outputs/add_entry_1_example_control_blocks.c

```diff
@@ -114,8 +114,8 @@ if (/* hdr->ipv4.isValid() */
                                 break;
                             case MAINCONTROLIMPL_IPV4_TBL_1_ACT_MAINCONTROLIMPL_SEND_NH: 
                                 {
-                                    hdr->ethernet.srcAddr = bpf_cpu_to_be64(value->u.MainControlImpl_send_nh.smac);
-                                                                        hdr->ethernet.dstAddr = ntohll(value->u.MainControlImpl_send_nh.dmac << 16);
+                                    storePrimitive64(&hdr->ethernet.srcAddr, 48, (bpf_cpu_to_be64(getPrimitive64(value->u.MainControlImpl_send_nh.smac, 48))));;
```

Comment:
Just a note, to avoid warnings cast the u8[6] pointers to `u8 *`

```suggestion
                                    storePrimitive64((u8 *)&hdr->ethernet.srcAddr, 48, (bpf_cpu_to_be64(getPrimitive64((u8 *)value->u.MainControlImpl_send_nh.smac, 48))));;
```

This goes to all the places where the code is passing a u8 array pointer to `storePrimitive64` or `getPrimitive64`